src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,231 +82,231 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
-            }
-        } else {
-	        integer i = 0;        
-	        for (SObject so : listNew) {
-	            Address__c addr = (Address__c)so;
-	            Address__c addrOld = (listOld != null) ? (Address__c)listOld[i] : null;
+   
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        if(!alreadyRunBeforeInsert || !alreadyRunBeforeUpdate || !alreadyRunAfterInsert || !alreadyRunAfterUpdate || !alreadyRunAfterDelete) {
+            //Turn off other address triggers
+            ADDR_Account_TDTM.turnOff();
+            ADDR_Contact_TDTM.turnOff();
+        
+	        // STRATEGY
+	        // when a default Address is created or modified, we need to consider three other objects we may need to update:
+	        // - the parent Account, if not null
+	        // - the parent Contact, if not null
+	        // - any other Addresses that should no longer be Default
+	        
+	        // map of parents, and their default Address
+	        Map<Id, Address__c> mapParentIdAddr = new Map<Id, Address__c>();
 	            
-	            // BEFORE INSERT
-	            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-	               // when an address is marked Default, we update its latest date fields
-	               if (addr.Default_Address__c) { 
-	                   addr.Latest_Start_Date__c = System.today();
-	                   addr.Latest_End_Date__c = null;
+	        // BEFORE INSERT
+	        if (!alreadyRunBeforeInsert && triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+		        for (SObject so : listNew) {
+		            Address__c addr = (Address__c)so;
+	                // when an address is created as Default, we update its latest date fields
+	                if (addr.Default_Address__c) {
+	                    addr.Latest_Start_Date__c = System.today();
+	                    addr.Latest_End_Date__c = null;
 	               }
-	            }
-	            
-	            // BEFORE UPDATE
-	            if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+		        }
+		        alreadyRunBeforeInsert = true;
+	        }
+	        
+	        // BEFORE UPDATE
+	        if (!alreadyRunBeforeUpdate && triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+	            Integer i = 0;
+		        for (SObject so : listNew) {
+		            Address__c addr = (Address__c)so;
+		            Address__c addrOld = listOld == null? null : (Address__c)listOld[i];
 	                if (addr.Default_Address__c != addrOld.Default_Address__c) {
-	                   // when an address changes Default, we update its latest date fields
+	                   // when a Default address changes, we update its latest date fields
 	                    if (addr.Default_Address__c) {
 	                       addr.Latest_Start_Date__c = System.today();
 	                       addr.Latest_End_Date__c = null;
 	                    } else {
+	                       UTIL_Debug.debug('****Address updated to not default');
```

Comment:
what if the user is just updating a non default address.  won't you remark its latest end date even though it wasn't changed from default to non-default?